### PR TITLE
[WC-2323]feat(app-events): add interval timer

### PR DESCRIPTION
**Pull request by gjulivan**
### Pull request type

New feature (non-breaking change which adds functionality)
---

### Description

adding interval feature to component load.

interval timer will execute after initial delay.

### CODE REVIEW

File: packages/pluggableWidgets/app-events-web/src/AppEvents.tsx [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
-    }, [attributeEvent]);
+    useActionTimer({
+        canExecute: onComponentLoad?.canExecute,
+        execute: onComponentLoad?.execute,
+        delay: componentLoadDelay,
+        interval: componentLoadRepeatInterval,
+        repeat: componentLoadRepeat
+    });
 
+    useActionTimer({
```
Comment (rahmanunver) [inline packages/pluggableWidgets/app-events-web/src/AppEvents.tsx:28]: Why do we have two calls for `useActionTimer`?

Comment (gjulivan) [reply packages/pluggableWidgets/app-events-web/src/AppEvents.tsx:28]: there are 2 events: component load and attribute change.

both has different useEffect parameters.
thus, 2 useEffect.

**Review by rahmanunver — APPROVED** [review]

File: packages/pluggableWidgets/app-events-web/src/hooks/timer.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+interface ActionTimerProps {
+    canExecute?: boolean;
+    execute?: () => void;
+    delay: number;
+    interval: number;
+    repeat: boolean;
+    attribute?: EditableValue;
+}
+
+export function useActionTimer(props: ActionTimerProps) {
```
Comment (iobuhov) [inline packages/pluggableWidgets/app-events-web/src/hooks/timer.ts:13]: Can we fix linter issues?

**Review by leonardomendix — APPROVED** [review]

File: packages/pluggableWidgets/app-events-web/src/hooks/timer.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+        return () => {
+            if (canExecute) {
+                if (repeat) {
+                    clearInterval(counter);
+                } else {
+                    clearTimeout(counter);
+                }
+            }
```
[the remark this answers is not in the document]
Comment (gjulivan) [reply packages/pluggableWidgets/app-events-web/src/hooks/timer.ts:45]: added as TODO on WC-2324

File: packages/pluggableWidgets/app-events-web/src/hooks/timer.ts (hunk 2 above)
Comment (gjulivan) [reply packages/pluggableWidgets/app-events-web/src/hooks/timer.ts:13]: added as TODO on WC-2324

**Pull request merged**